catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/WhenBrickTest.java

```diff
@@ -0,0 +1,167 @@
+
+	@Before
+	public void setUp() throws Exception {
+
+		Script script = BrickTestUtils.createProjectAndGetStartScript("WhenBrickTest");
+		script.addBrick(new PlaceAtBrick());
+
+		whenBrickMap.put(R.string.brick_when_started,WhenStartedBrick.class);
+		whenBrickMap.put(R.string.brick_when_becomes_true,WhenConditionBrick.class);
+		whenBrickMap.put(R.string.brick_when_cloned,WhenClonedBrick.class);
+		whenBrickMap.put(R.string.brick_when_background,WhenBackgroundChangesBrick.class);
+		whenBrickMap.put(R.string.brick_when_touched,WhenTouchDownBrick.class);
+		whenBrickMap.put(R.string.brick_broadcast_receive,BroadcastReceiverBrick.class);
+		whenBrickMap.put(R.string.brick_collision_receive,CollisionReceiverBrick.class);
+
+		whenBrickList.add(R.string.brick_when_started);
+		whenBrickList.add(R.string.brick_when_becomes_true);
+		whenBrickList.add(R.string.brick_when_cloned);
+		whenBrickList.add(R.string.brick_when_background);
+		whenBrickList.add(R.string.brick_when_touched);
+		whenBrickList.add(R.string.brick_broadcast_receive);
+		whenBrickList.add(R.string.brick_collision_receive);
+
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testWhenBrick() {
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_place_at);
+
+		for(int i = 0; i < whenBrickList.size(); i++)
+		{
+			addWhenBrickAndMoveToTop(whenBrickMap.get(whenBrickList.get(i)),R.string.category_event);
+			addBrickAndMoveToTop(PlaceAtBrick.class, R.string.category_motion);
+
+			onBrickAtPosition(0).checkShowsText(whenBrickList.get(i));
+			onBrickAtPosition(1).checkShowsText(R.string.brick_place_at);
+
+			onBrickAtPosition(0).performDragNDrop(BrickCoordinatesProvider.DOWN_TO_BOTTOM);
+			onBrickAtPosition(0).checkShowsText(whenBrickList.get(i));
+
+			onBrickAtPosition(1).performDragNDrop(BrickCoordinatesProvider.DOWN_TO_BOTTOM);
+			onBrickAtPosition(3).checkShowsText(R.string.brick_place_at);
+
+			onBrickAtPosition(3).performDeleteBrick();
+			onBrickAtPosition(0).performDeleteBrick();
+
+			onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+			onBrickAtPosition(1).checkShowsText(R.string.brick_place_at);
+		}
+	}
+
+	public void addBrickAndMoveToTop(Class<?> brickHeaderClass, int brickCategoryId) {
+		onView(withId(R.id.button_add))
+				.perform(click());
+		onData(allOf(is(instanceOf(String.class)), is(UiTestUtils.getResourcesString(brickCategoryId))))
+				.inAdapterView(BrickCategoryListMatchers.isBrickCategoryView())
+				.perform(click());
+		onData(is(instanceOf(brickHeaderClass))).inAdapterView(BrickPrototypeListMatchers.isBrickPrototypeView())
+				.perform(click());
+
+		onBrickAtPosition(1).performDragNDrop(BrickCoordinatesProvider.UP_TO_TOP);
+	}
+
+	public void addWhenBrickAndMoveToTop(Class<?> brickHeaderClass, int brickCategoryId) {
```

Comment:
since  int brickCategoryId is always the same, move it either into this function, or make it a constant. no need to clutter the function signature